src/FormField/ScopeBuilder.php

```diff
@@ -0,0 +1,595 @@
+<?php
+
+namespace atk4\ui\FormField;
+
+use atk4\data\Model;
+use atk4\ui\Exception;
+use atk4\ui\Template;
+use atk4\data\Field;
+use atk4\data\Model\Scope\Condition;
+use atk4\data\Model\Scope\Scope;
+use atk4\data\Model\Scope\AbstractScope;
+
+class ScopeBuilder extends Generic
+{
+    /**
+     * Field type specific options.
+     *
+     * @var array
+     */
+    public $options = [
+        'enum' => [
+            'limit' => 250
+        ]
+    ];
+    /**
+     * Max depth of nested conditions allowed.
+     * Corresponds to VueQueryBulder maxDepth
+     *
+     * @var integer
+     */
+    public $maxDepth = 10;
+
+    /**
+     * Fields to use for creating the rules
+     *
+     * @var array
+     */
+    public $fields = [];
+
+    /**
+     * The template needed for the scopebuilder view.
+     *
+     * @var Template
+     */
+    public $scopeBuilderTemplate;
+
+    /**
+     * List of delimiters for auto-detection in order of priority
+     *
+     * @var array
+     */
+    public static $listDelimiters = [';', ',', '|'];
```

Comment:
When `|` makes sense? I think we do not need it at all.

Reply:
Aiming at more versatility. Some users are used to using | for a list delimiter (from DynamicsNAV as one example)

Reply:
I think better to not make default because it is confusing with `||`.

Reply:
@georgehristov can we remove `|` from defaults?

Reply:
Done